aspects/aspects.go

```diff
@@ -306,57 +306,127 @@ func (a *Aspect) Set(databag DataBag, request string, value interface{}) error {
 		return a.bundle.schema.Validate(data)
 	}
 
+	return notFoundErrorFrom(a, request, fmt.Errorf("field not found"))
+}
+
+func notFoundErrorFrom(a *Aspect, request string, err error) *NotFoundError {
 	return &NotFoundError{
 		Account:    a.bundle.Account,
 		BundleName: a.bundle.Name,
 		Aspect:     a.Name,
 		Field:      request,
-		Cause:      "field not found",
+		Cause:      err.Error(),
 	}
 }
 
-// Get returns the aspect value identified by the name. If either the named aspect
-// or the corresponding value can't be found, a NotFoundError is returned.
-func (a *Aspect) Get(databag DataBag, request string, value interface{}) error {
+// Get returns the aspect value identified by the request. If either the named
+// aspect or the corresponding value can't be found, a NotFoundError is returned.
+func (a *Aspect) Get(databag DataBag, request string, value *interface{}) error {
+	matches, err := a.matchGetRequest(request)
+	if err != nil {
+		return err
+	}
+
+	// there's an exact match, return only that one
+	if len(matches) == 1 && matches[0].exact() {
+		var val interface{}
+		if err := databag.Get(matches[0].storagePath, &val); err != nil {
+			var pathErr PathNotFoundError
+			if errors.As(err, &pathErr) {
+				return notFoundErrorFrom(a, request, pathErr)
+			}
+			return err
+		}
+
+		*value = map[string]interface{}{request: val}
+		return nil
+	}
+
+	var results []interface{}
+	for _, match := range matches {
+		var val interface{}
+		if err := databag.Get(match.storagePath, &val); err != nil {
+			var pathErr PathNotFoundError
+			if errors.As(err, &pathErr) {
+				return notFoundErrorFrom(a, request, pathErr)
```

Comment:
I think it probably makes sense to only return not-found if no matches have values and otherwise return what could be found